src/client/Microsoft.Identity.Client/Internal/Broker/BrokerInteractiveRequestComponent.cs

```diff
@@ -59,44 +59,17 @@ public async Task<MsalTokenResponse> FetchTokensAsync(CancellationToken cancella
                 }
 
                 _logger.Info(LogMessages.AddBrokerInstallUrlToPayload);
-                BrokerPayload[BrokerParameter.BrokerInstallUrl] = _optionalBrokerInstallUrl;
+                Broker.HandleInstallUrl(_optionalBrokerInstallUrl);                
             }
 
-            var tokenResponse = await SendAndVerifyResponseAsync().ConfigureAwait(false);
-            return tokenResponse;
-        }
+            var tokenResponse = await Broker.AcquireTokenInteractiveAsync(
+                _authenticationRequestParameters, 
+                _interactiveParameters)
+                .ConfigureAwait(false);
 
-        private async Task<MsalTokenResponse> SendAndVerifyResponseAsync()
-        {
-            CreateRequestParametersForBroker();
+            ValidateResponseFromBroker(tokenResponse);
 
-            MsalTokenResponse msalTokenResponse =
-                await Broker.AcquireTokenUsingBrokerAsync(BrokerPayload).ConfigureAwait(false);
```

Comment:
The main improvement is getting rid of the weakly typed dictionary and using strongly typed objects.